### DISCUSSION TITLE
better way to locally test a renovate configuration?

### DISCUSSION THREAD

**Discussion by sjiekak** [renovatebot/renovate#17131] [Request Help] [answered]
We are using renovate in our organisation, and changes in master **must** be reviewed before merge.
Currently, we have found no better way to test the config than : 

Done once : 
- select a test repository
- create a separate branch `renovate_test` which is going to be used to test
- in the main branch renovate file, specify `renovate_test` branch as a base branch.
- commit + push

Each time we like to update:
- rebase `renovate_test` to `master`
- commit + push changes to `renovate_test`
- validate the result locally by making a dry run with the renovate config in `renovate_test` active  

```
docker run renovate/renovate \
   --platform github\
   --token=$GITHUB_TOKEN\
   --print-config=true\
   --force-cli=true\
   --dry-run=full\
   --use-base-branch-config=merge\
     $GITHUB_REPOSITORY
```

Is there any simpler way to test our config on an renovate-enabled repository? Preferably without having to modify remote branches ?

**Comment by PhilipAbed** [comment]
do u mean renovate-config-validator?

**Reply by sjiekak** [reply]
Not really. I would like to review the changes to be applied.

**Reply by StinkyLord** [reply]
so you need to test if your config actually works, without running dry run at least?

**Reply by sjiekak** [reply]
Rather I would like to run the dry run on a custom config, without having to commit to remote github beforehand. Right now I have to commit the custom config, run locally with dry run, look at the planned changes, modify the config, commit, etc...

**Reply by sjiekak** [reply]
The same way originatorConfig can be used to set up the repo the first time, I would like to inject a configuration locally.

**Reply by PhilipAbed** [reply]
That is possible yes. Put all ur config in the global config.js config file then pass it to the docker using -v as a shared file (path should be to /app inside docker)

**Comment by PhilipAbed** [comment, marked as answer]
i use `-v "C:\path\to\config.js:/usr/src/app/config.js"`
on my docker command
and then 
my file looks ike this: 
```
module.exports = {
  token: 'GITHUB_COM_TOKEN',
  platform: 'github',
  repositories: ['PhilipAbed/renovate'], 
  includeForks: true,
  onboarding: false,
  gitAuthor: 'Self-hosted Renovate Bot <xyz@gmail.com>',
};
```

i believe u can do the same with `.json`

**Reply by sjiekak** [reply]
@PhilipAbed thank you this is what I was looking for

**Comment by btxbtx** [comment]
For testing, I've downloaded renovate via `npm` and run it directly from the directory where my config is.
